modules/apps/segments/segments-test/src/testIntegration/java/com/liferay/segments/web/internal/portlet/action/test/EditSegmentsEntryMVCRenderCommandTest.java

```diff
@@ -83,6 +95,58 @@ public void setUp() throws Exception {
 		_group = GroupTestUtil.addGroup();
 	}
 
+	@Test(expected = ExpressionVisitException.class)
+	public void testGetDataWithErrorMessage() throws Exception {
+		MockLiferayPortletRenderRequest mockLiferayPortletRenderRequest =
+			_getMockLiferayPortletRenderRequests();
+
+		MockLiferayPortletRenderResponse mockLiferayPortletRenderResponse =
+			new MockLiferayPortletRenderResponse();
+
+		mockLiferayPortletRenderRequest.setAttribute(
+			WebKeys.USER, TestPropsValues.getUser());
+
+		_expandoTable = ExpandoTestUtil.addTable(
+			PortalUtil.getClassNameId(User.class), "CUSTOM_FIELDS");
+
+		ExpandoColumn expandoColumn = _addExpandoColumn(
+			_expandoTable, RandomTestUtil.randomString(),
+			ExpandoColumnConstants.BOOLEAN,
+			ExpandoColumnConstants.INDEX_TYPE_KEYWORD);
+
+		SegmentsEntry segmentsEntry = _addSegmentEntry(
+			String.format(
+				"(customField/%s eq %s)", _encodeName(expandoColumn),
+				StringPool.TRUE));
+
+		_expandoColumnLocalService.deleteColumn(expandoColumn);
+
+		mockLiferayPortletRenderRequest.setParameter(
+			"segmentsEntryId",
+			String.valueOf(segmentsEntry.getSegmentsEntryId()));
+
+		_mvcRenderCommand.render(
+			mockLiferayPortletRenderRequest, mockLiferayPortletRenderResponse);
+
+		Map<String, Object> data = ReflectionTestUtil.invoke(
+			mockLiferayPortletRenderRequest.getAttribute(
+				"EDIT_SEGMENTS_ENTRY_DISPLAY_CONTEXT"),
+			"getData", new Class<?>[0]);
+
+		Map<String, Object> props = (Map<String, Object>)data.get("props");
+
+		JSONArray contributorsJSONArray = (JSONArray)props.get("contributors");
+
+		JSONObject initialQueryJSONObject = contributorsJSONArray.getJSONObject(
+			0
+		).getJSONObject(
+			"initialQuery"
+		);
+
+		Assert.assertEquals(
+			"true", initialQueryJSONObject.getString("invalidSegmenterror"));
```

Comment:
I think the right value is invalidSegmentError

Reply:
But as your are expecting invalidSegmenterror, it doesn't make sense to assert the invalidSegmenterror, this code is not going to be executed until the FF is deleted.

Reply:
Thanks!

Reply:
Yes, because with the FF to false, the exception will be thrown and the Test is expecting that exception and this part of the test won't be exectued. When we'll delete the FF and we'll change the behaviour of the test (delete the @exception expected) this we'll be needed.

Reply:
thanks!